src/providers/useRegistryAddress.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  CHILD_GAUGE_FACTORY_NAME,
+  useMasterRegistry,
+} from "../hooks/useContract"
+import { UseQueryResult, useQuery } from "@tanstack/react-query"
+import { ChainId } from "../constants"
+import { parseBytes32String } from "ethers/lib/utils"
+import { useActiveWeb3React } from "../hooks"
+
+export type RegistryAddress = Partial<Record<string, string>>
+
+export const useRegistryAddress = (): UseQueryResult<RegistryAddress> => {
+  const { chainId } = useActiveWeb3React()
+  const masterRegistry = useMasterRegistry()
+
+  return useQuery(["registryAddress"], async () => {
+    if (!masterRegistry) {
+      throw new Error("Master Registry not available. Unable to retrieve contract addresses")
+    }
+
+    if (chainId !== ChainId.MAINNET && chainId !== ChainId.HARDHAT) {
+      const childGaugeFactoryAddress =
+        await masterRegistry.resolveNameToLatestAddress(
+          CHILD_GAUGE_FACTORY_NAME,
+        )
+      const addresses: Partial<Record<string, string>> = {}
+      addresses[parseBytes32String(CHILD_GAUGE_FACTORY_NAME)] =
```

Comment:
thoughts on changing `CHILD_GAUGE_FACTORY_NAME` to `CHILD_GAUGE_FACTORY_NAME_BYTES32` or storing it as the parsed name and only formatting it for the contract call?

Reply:
+1 on this.